From 73b6eae583f44e278e19489a411f9c1e22d530fc Mon Sep 17 00:00:00 2001
Message-Id: <73b6eae583f44e278e19489a411f9c1e22d530fc.1648731712.git.geliang.tang@suse.com>
From: Colin Ian King <colin.i.king@gmail.com>
Date: Tue, 7 Dec 2021 22:47:18 +0000
Subject: [PATCH] bpf: Remove redundant assignment to pointer t
Git-commit: 73b6eae583f44e278e19489a411f9c1e22d530fc
Patch-mainline: v5.17-rc1
References: bsn#127
Modified-by-SEL: No


The pointer t is being initialized with a value that is never read. The
pointer is re-assigned a value a littler later on, hence the initialization
is redundant and can be removed.

Signed-off-by: Colin Ian King <colin.i.king@gmail.com>
Signed-off-by: Andrii Nakryiko <andrii@kernel.org>
Link: https://lore.kernel.org/bpf/20211207224718.59593-1-colin.i.king@gmail.com
Signed-off-by: Geliang Tang <geliang.tang@suse.com>
---
 kernel/bpf/btf.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/kernel/bpf/btf.c b/kernel/bpf/btf.c
index 01b47d4df3ab..27b7de538697 100644
--- a/kernel/bpf/btf.c
+++ b/kernel/bpf/btf.c
@@ -837,7 +837,7 @@ static const char *btf_show_name(struct btf_show *show)
 	const char *ptr_suffix = &ptr_suffixes[strlen(ptr_suffixes)];
 	const char *name = NULL, *prefix = "", *parens = "";
 	const struct btf_member *m = show->state.member;
-	const struct btf_type *t = show->state.type;
+	const struct btf_type *t;
 	const struct btf_array *array;
 	u32 id = show->state.type_id;
 	const char *member = NULL;
-- 
2.34.1

